packages/react/src/components/Tree/Tree.tsx

```diff
@@ -60,14 +78,47 @@ class Tree extends UIComponent<ReactProps<TreeProps>> {
     accessibility: defaultBehavior,
   }
 
+  static autoControlledProps = ['activeIndex']
+
+  getInitialAutoControlledState({ defaultActiveIndex, exclusive }) {
+    return {
+      activeIndex: defaultActiveIndex === undefined ? (exclusive ? -1 : []) : defaultActiveIndex,
+    }
+  }
+
+  computeNewIndex = index => {
+    const { activeIndex } = this.state
+    const { exclusive } = this.props
+
+    if (exclusive) return index === activeIndex ? -1 : index
+    // check to see if index is in array, and remove it, if not then add it
+    return _.includes(activeIndex, index) ? _.without(activeIndex, index) : [...activeIndex, index]
+  }
+
+  handleTreeItemOverrides = predefinedProps => ({
+    onOpenChange: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+      const { index } = treeItemProps
+      this.trySetState({
+        activeIndex: this.computeNewIndex(index),
+      })
+      _.invoke(predefinedProps, 'onOpenChange', e, treeItemProps)
+    },
+  })
+
   renderContent() {
-    const { items, renderItemTitle } = this.props
+    const { items, renderItemTitle, exclusive } = this.props
```

Comment:
```suggestion
    const { items, renderItemTitle, exclusive } = this.props
    const { activeIndex } = this.state
```

Let's desctruct `activeIndex `